Addition of io_k8s_repos and removal of vendor deps

chases2 opened the pull request:
Allows use of @io_k8s_repo_infra//hack commands in bazel
Also runs automatic cleanup/update command
@io_k8s_repo_infra//hack:update-deps

Useful cleanup towards #2140

fejta commented on testgrid/istio.gen.yaml:
```diff
@@ -1,5509 +0,0 @@
-dashboard_groups:
```
Is this intentional?

chases2 replied:
No; this change should be in a different PR. This is what I was working on before

chases2 committed:
Addition of io_k8s_repos and removal of vendor deps

Allows use of @io_k8s_repo_infra//hack commands in bazel
Also runs automatic cleanup/update command
@io_k8s_repo_infra//hack:update-deps

chases2 committed:
Run make gen

fejta reviewed (approved):
/hold
/assign @clarketm @howardjohn 
fyi
